### PR TITLE
Add github_app_installation_repository resource

### PR DESCRIPTION
**Pull request by k24dizzle**
Example Usage:
```
resource "github_app_installation_repository" "example" {
    repository      = "testrepo"
    installation_id = "1234567"
}
```

Gave https://github.com/integrations/terraform-provider-github/issues/687 a shot. However, while this implementation works in my local testing (I was able to install an existing app to existing repos), this PR is incomplete.

I was going to write acceptance tests for this PR, but realized that there is no way to spin up an organization that already has an app installed. We would need to write a corresponding resource that installs github apps in an organization, but I don't believe that there is an easy way to do that through the Github API.

Any ideas on potential next steps/feedback would be greatly appreciated!

### CODE REVIEW

**Comment by jcudit** [comment]
Looking great so far!  A couple of suggestions ahead of merging this:
- Since this is currently difficult to acceptance test, we could benefit from a new example in the `examples/` directory
- We'll need to document this similar to other resources (see `website/docs/r` and `website/github.erb`)

❤️  the direction here.  Feels like we're closer to shipping https://github.com/integrations/terraform-provider-github/issues/509 and https://github.com/integrations/terraform-provider-github/issues/514 with this feature in place.  Thanks for driving this!

**Comment by k24dizzle** [comment]
Thanks for the comments @jcudit, went ahead and implemented your feedback. Let me know if I can do anything else/if you have more feedback!

**Comment by k24dizzle** [comment]
Hi @jcudit I was wondering if there were any updates to this PR, feel free to let me know if there's any more work to be done on this

**Comment by jcudit** [comment]
Thanks for the ping 🙇🏾 .  A few of us are going to pair on reviews this week.  Aiming to get this one tested out during the session and will likely ship with the next release.

**Comment by jcudit** [comment]
A few of us are going to pair on reviews this week.  Aiming to get this one tested out during the session and return here with next steps.  🤞🏾 that things just work and we can get to shipping.

File: github/resource_github_app_installation_repository.go [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+	"github.com/google/go-github/v32/github"
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+)
+
+func resourceGithubAppInstallationRepository() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceGithubAppInstallationRepositoryCreate,
+		Read:   resourceGithubAppInstallationRepositoryRead,
+		Delete: resourceGithubAppInstallationRepositoryDelete,
```
Comment (jcudit) [inline github/resource_github_app_installation_repository.go:16]: 🤔 would we benefit from an `Update` function here as well? Or is creating a new App installation usually non-disruptive?

File: website/docs/r/app_installation_repository.html.markdown [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+# Create a repository.
+resource "github_repository" "some_repo" {
+  name = "some-repo"
+}
+
+resource "github_app_installation_repository" "some_app_repo" {
+  # The installation id of the app (in the organization).
+  installation_id    = "1234567"
+  repository         = "${github_repository.some_repo.name}"
+}
```
Comment (jcudit) [inline website/docs/r/app_installation_repository.html.markdown:32]: Can we get a test that exercises this configuration?  Open to updates in HCL syntax as well.

**Review by jcudit — COMMENTED** [review]
Excited to see this one land.  Thanks for getting it this far and being patient on the feedback 🙇🏾 .

File: website/docs/r/app_installation_repository.html.markdown (hunk 2 above)
Comment (jcudit) [reply website/docs/r/app_installation_repository.html.markdown:32]: 🤦🏾  I now see an earlier comment where I requested an example over an acceptance test.  I'll give testing a shot and plan to move forward with this with just the example if the test does not pan out.

**Review by jcudit — APPROVED** [review]
This tested well locally and I've got eaaaf6561722b11dd8fe8c86e0bbfee7733874c3 queued up to commit after this merges as-is.  Thanks again for this contribution and your patience up to this point @k24dizzle.

**Pull request merged**